pkg/cloudprovider/provider/vsphere/provider.go

```diff
@@ -240,6 +254,12 @@ func (p *provider) Create(machine *v1alpha1.Machine, userdata string) (instance.
 		return nil, fmt.Errorf("failed to get virtual machine object: %v", err)
 	}
 
+	// Map networks
+	err = updateNetworkForVM(context.TODO(), virtualMachine, config.TemplateNetName, config.VMNetName)
```

Comment:
Please only update the network if `VMNetName != ""` to not make this a required config option